pennylane/devices/device_api.py

```diff
@@ -335,7 +348,9 @@ def execute_fn(tapes):
         transform_program = self.preprocess_transforms(execution_config)
         return transform_program, execution_config
 
-    def setup_execution_config(self, config: Optional[ExecutionConfig] = None) -> ExecutionConfig:
+    def setup_execution_config(
+        self, config: Optional[ExecutionConfig] = None, tape: Optional[QuantumScript] = None
```

Comment:
Given we use `circuit` on the other methods, I'm actually wondering if we should use `circuit` here too.